app/utils/base_client2.js

```diff
@@ -19,17 +19,14 @@ function retryOnEconnreset (err) {
   return err && _.includes(RETRIABLE_ERRORS, err.code)
 }
 
-/**
- * @type {https.Agent}
- */
-const httpsAgent = new https.Agent(agentOptions)
-
 if (process.env.DISABLE_INTERNAL_HTTPS !== 'true') {
-  httpsAgent.ca = customCertificate.getCertOptions()
+  agentOptions.ca = customCertificate.getCertOptions()
 } else {
   logger.warn('DISABLE_INTERNAL_HTTPS is set.')
 }
 
+const httpsAgent = new https.Agent(agentOptions)
+
```

Comment:
I tried to get rid of baseClient but seems like a bigger change. Hence the `base_client2.js` which is the same supporting `pact` stuff. This is new and was not used before. On my previous commit I had this options setting up wrong way around.